source

```diff
@@ -107224,6 +107259,121 @@ document.querySelector("button").addEventListener("click", bound);
+    <var>settings</var>'s <span data-x="environment settings object's realm">realm</span>,
+    <var>script</var>).</p>
+
+    <p class="note">Passing <var>script</var> as the last parameter here ensures
+    <var>result</var>.[[HostDefined]] will be <var>script</var>.</p>
+   </li>
+
+   <li>
+    <p>If <var>result</var> is a <span>list</span> of errors, then:</p>
+
+    <ol>
+     <li><p>Set <var>script</var>'s <span data-x="concept-script-parse-error">parse error</span> to
+     <var>result</var>[0].</p></li>
+
+     <li><p>Return <var>script</var>.</p></li>
+    </ol>
+   </li>
+
+   <li id="validate-requested-module-specifiers">
+    <p><span data-x="list iterate">For each</span> <span>ModuleRequest record</span>
+    <var>requested</var> of <var>result</var>.[[RequestedModules]]:</p>
+
+    <ol>
+     <li>
+     <p>If <var>requested</var>.[[Attributes]] contains a <span>Record</span> <var>entry</var>
+     such that <var>entry</var>.[[Key]] is not "<code data-x="">type</code>", then:</p>
+
+     <ol>
+      <li><p>Let <var>error</var> be a new <code>SyntaxError</code> exception.</p></li>
+
+      <li><p>Set <var>script</var>'s <span data-x="concept-script-parse-error">parse error</span>
+      to <var>error</var>.</p></li>
+
+      <li><p>Return <var>script</var>.</p></li>
+     </ol>
+
+     <p class="note">The JavaScript specification re-performs this validation when loading
+     <var>script</var>'s dependencies. It is duplicated here to skip unnecessary work given that,
+     in case of an invalid attribute key, loading the module graph would fail anyway. Additionally,
+     this makes the errors reported for invalid static imports consistent with dynamic imports, for
+     which invalid attributes are reported before calling into HTML and thus before validating the
+     imported specifier.</p>
+     </li>
+
+     <li><p><span>Resolve a module specifier</span> given <var>script</var> and
+     <var>requested</var>.[[Specifier]], catching any exceptions.</p></li>
+
+     <li>
+     <p>If the previous step threw an exception, then:</p>
+
+     <ol>
+      <li><p>Set <var>script</var>'s <span data-x="concept-script-parse-error">parse error</span>
+      to that exception.</p></li>
+
+      <li><p>Return <var>script</var>.</p></li>
+     </ol>
+    </li>
+
+    <li><p>Let <var>moduleType</var> be the result of running the <span>module type from module
+    request</span> steps given <var>requested</var>.</p></li>
+
+    <li>
+     <p>If the result of running the <span>module type allowed</span> steps given
+     <var>moduleType</var> and <var>settings</var> is false, then:</p>
+
+     <ol>
+      <li><p>Let <var>error</var> be a new <code>TypeError</code> exception.</p></li>
+
+      <li><p>Set <var>script</var>'s <span data-x="concept-script-parse-error">parse error</span>
+      to <var>error</var>.</p></li>
+
+       <li><p>Return <var>script</var>.</p></li>
+      </ol>
+     </li>
```

Comment:
And thus we know that here `moduleType` is always `"javascript-or-wasm"`, and _module type allowed_ always returns true.